slay/example_workflow/workflow.py

```diff
@@ -0,0 +1,203 @@
+        )
+
+        self._generate_args = {
+            "max_new_tokens": 512,
+            "temperature": 1.0,
+            "top_p": 0.95,
+            "top_k": 50,
+            "repetition_penalty": 1.0,
+            "no_repeat_ngram_size": 0,
+            "use_cache": True,
+            "do_sample": True,
+            "eos_token_id": self._tokenizer.eos_token_id,
+            "pad_token_id": self._tokenizer.pad_token_id,
+        }
+
+    def run(self, data: str) -> str:
+        import torch
+
+        formatted_prompt = f"[INST] {data} [/INST]"
+        input_ids = self._tokenizer(
+            formatted_prompt, return_tensors="pt"
+        ).input_ids.cuda()
+        with torch.no_grad():
+            output = self._model.generate(inputs=input_ids, **self._generate_args)
+            result = self._tokenizer.decode(output[0])
+        return result
+
+
+class MistralP(Protocol):
+    def __init__(self, context: slay.Context) -> None:
+        ...
+
+    def run(self, data: str) -> str:
+        ...
+
+
+class TextToNum(slay.ProcessorBase):
+    default_config = slay.Config(image=IMAGE_COMMON)
+
+    def __init__(
+        self,
+        context: slay.Context = slay.provide_context(),
+        mistral: MistralP = slay.provide(MistralLLM),
+    ) -> None:
+        super().__init__(context)
+        self._mistral = mistral
+
+    def run(self, data: str) -> int:
+        number = 0
+        generated_text = self._mistral.run(data)
+        for char in generated_text:
+            number += ord(char)
+
+        return number
+
+
+class Workflow(slay.ProcessorBase):
+    default_config = slay.Config(image=IMAGE_COMMON)
+
+    def __init__(
+        self,
+        context: slay.Context = slay.provide_context(),
+        data_generator: GenerateData = slay.provide(GenerateData),
+        splitter: shared_processor.SplitText = slay.provide(shared_processor.SplitText),
+        text_to_num: TextToNum = slay.provide(TextToNum),
+    ) -> None:
+        super().__init__(context)
+        self._data_generator = data_generator
+        self._data_splitter = splitter
+        self._text_to_num = text_to_num
+
+    async def run(self, length: int, num_partitions: int) -> tuple[int, str, int]:
+        data = self._data_generator.run(length)
+        text_parts, number = await self._data_splitter.run(data, num_partitions)
+        value = 0
+        for part in text_parts:
+            value += self._text_to_num.run(part)
+        return value, data, number
+
+
+if __name__ == "__main__":
+    import logging
+
+    from slay import utils
+
+    # from slay.truss_compat import deploy
+
+    root_logger = logging.getLogger()
+    root_logger.setLevel(logging.INFO)
+    log_format = "%(levelname).1s%(asctime)s %(filename)s:%(lineno)d] %(message)s"
+    date_format = "%m%d %H:%M:%S"
+    formatter = logging.Formatter(fmt=log_format, datefmt=date_format)
+    for handler in root_logger.handlers:
+        handler.setFormatter(formatter)
+
```

Comment:
why is this logger config needed?

Reply:
It's not technically needed, I just wanted to log like this for my dev loop.